File: ChessApp.Server/Models/Game.cs
namespace ChessApp.Server.Models
{
    public class Game
    {
        public string GameId { get; set; } = Guid.NewGuid().ToString();
        public required string CreatedBy { get; set; }
        public TimeOnly CreatedAt { get; set; } = TimeOnly.FromDateTime(DateTime.Now);
        public Player? PlayerWhite { get; set; }
        public Player? PlayerBlack { get; set; }
        public List<GameTurn> Turns { get; set; } = new List<GameTurn>();
        public GameStatus Status { get; set; } = GameStatus.Waiting;
        public GameType Type { get; set; } = GameType.Standard;
        public int TimeIncrementPerMove { get; set; } = 0;
        
        public Game() { }
    }

    public enum GameType
    {
        Bullet,
        Blitz,
        Rapid,
        Standard,
        Classical
    }

    public enum GameStatus{
        Waiting,
        Started,
        Abandoned,
        Ended
    }
}

File: chessApp.client/src/types/Game.ts
import { GameTurn } from "./GameTurn";
import { Player } from "./Player"

export type Game = {
    readonly gameId: string;
    readonly createdBy: string;
    readonly createdAt: string;
    readonly playerWhite: Player | null;
    readonly playerBlack: Player | null;
    turns: GameTurn[];
    status: GameStatus;
    type: GameType;
    timeIncrementPerMove: number;
}

export enum GameStatus {
    Waiting = 0,
    Started = 1,
    Abandoned = 2,
    Ended = 3
}

export enum GameType {
    Bullet = 0,
    Blitz = 1,
    Rapid = 2,
    Standard = 3,
    Classical = 4
}